### PR TITLE
Lmb 1111 | improve linked mandatarissen migrations

### PR DESCRIPTION
**Pull request by JonasVanHoof**
## Description

Migrations do not always run. Improve the queries so they are less heavy.

## How to test

 - Start from prod db dump
- Let the migrations run and see if you have any hickups while migrating 

## Links to other PR's

- https://github.com/lblod/app-lokaal-mandatenbeheer/pull/316

### CODE REVIEW

File: config/migrations/2024/20241205154801-link-not-linked-mandatarissen.sparql [hunk 1]
```diff
@@ -13,32 +13,34 @@ INSERT {
...
+    ?linkedMandataris mandaat:start ?mandatarisStart.
+
     ?mandataris mandaat:isBestuurlijkeAliasVan ?persoon.
+    ?linkedMandataris mandaat:isBestuurlijkeAliasVan ?persoon.
+
     ?mandataris org:hasMembership / org:organisation ?fractie.
+    ?linkedMandataris org:hasMembership / org:organisation ?fractie.
+
+    ?mandaat ^org:hasPost / lmb:heeftBestuursperiode <http://data.lblod.info/id/concept/Bestuursperiode/96efb929-5d83-48fa-bfbb-b98dfb1180c7>. # 2024 - heden
+    ?lmandaat ^org:hasPost / lmb:heeftBestuursperiode <http://data.lblod.info/id/concept/Bestuursperiode/96efb929-5d83-48fa-bfbb-b98dfb1180c7>. # 2024 - heden
```
Comment (JonasVanHoof) [inline config/migrations/2024/20241205154801-link-not-linked-mandatarissen.sparql:33]: *Directly use periode uri

File: config/migrations/2024/20241205154801-link-not-linked-mandatarissen.sparql [hunk 2]
```diff
@@ -13,32 +13,34 @@ INSERT {
...
     }
-    ?lmandaat org:role bestuursfunctieCode:5ab0e9b8a3b2ca7c5e000015. # Lid RMW
-    ?lmandaat ^org:hasPost ?lbestuursorgaan.
-    ?lbestuursorgaan mandaat:isTijdspecialisatieVan ?lbestuursorgaanInTijd.
-    ?lbestuursorgaan lmb:heeftBestuursperiode ?bestuursperiode.
-  }
 
-  FILTER (?mandataris != ?linkedMandataris)
-  BIND (<http://data.lblod.info/id/concept/Bestuursperiode/96efb929-5d83-48fa-bfbb-b98dfb1180c7> AS ?bestuursperiode) # 2024 - heden
+    FILTER (?mandatarisEinde = ?mandatarisEinde2)
```
Comment (JonasVanHoof) [inline config/migrations/2024/20241205154801-link-not-linked-mandatarissen.sparql:43]: *Filter with optional for end dates

File: config/migrations/2024/20241205154802-link-not-linked-mandatarissen.sparql [hunk 3]
```diff
@@ -13,32 +13,34 @@ INSERT {
 WHERE {
   GRAPH ?g {
     ?mandataris org:holds ?mandaat.
+    ?mandaat org:role bestuursfunctieCode:5ab0e9b8a3b2ca7c5e000012. # Voorzitter Gemeenteraad
+
+    ?linkedMandataris org:holds ?lmandaat.
+    ?lmandaat org:role bestuursfunctieCode:5ab0e9b8a3b2ca7c5e000016. # Voorzitter RMW
+
+    FILTER (?mandataris != ?linkedMandataris)
```
Comment (JonasVanHoof) [inline config/migrations/2024/20241205154802-link-not-linked-mandatarissen.sparql:21]: * Move this to the top so non interesting subjects are filtered out first

File: config/migrations/2024/20241205154802-link-not-linked-mandatarissen.sparql [hunk 4]
```diff
@@ -13,32 +13,34 @@ INSERT {
...
+    ?linkedMandataris mandaat:start ?mandatarisStart.
+
     ?mandataris mandaat:isBestuurlijkeAliasVan ?persoon.
+    ?linkedMandataris mandaat:isBestuurlijkeAliasVan ?persoon.
+
     ?mandataris org:hasMembership / org:organisation ?fractie.
+    ?linkedMandataris org:hasMembership / org:organisation ?fractie.
+
+    ?mandaat ^org:hasPost / lmb:heeftBestuursperiode <http://data.lblod.info/id/concept/Bestuursperiode/96efb929-5d83-48fa-bfbb-b98dfb1180c7>. # 2024 - heden
+    ?lmandaat ^org:hasPost / lmb:heeftBestuursperiode <http://data.lblod.info/id/concept/Bestuursperiode/96efb929-5d83-48fa-bfbb-b98dfb1180c7>. # 2024 - heden
```
Comment (JonasVanHoof) [inline config/migrations/2024/20241205154802-link-not-linked-mandatarissen.sparql:33]: *fix check on period + use property paths

File: config/migrations/2024/20241205154805-link-not-linked-mandatarissen.sparql [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      ?mandataris mandaat:einde ?mandatarisEinde.
+    }
+
+    OPTIONAL {
+      ?linkedMandataris mandaat:einde ?mandatarisEinde2.
+    }
+
+    FILTER (?mandatarisEinde = ?mandatarisEinde2)
+  }
+  ?g ext:ownedBy ?something .
```
Comment (JonasVanHoof) [inline config/migrations/2024/20241205154805-link-not-linked-mandatarissen.sparql:45]: *Only check the bestuurseenheid graphs

**Review by JonasVanHoof — COMMENTED** [review]
Improvements done in the queries

**Review by Rahien — APPROVED** [review]

**Review by Rahien — APPROVED** [review]

**Pull request merged**